### PR TITLE
kpb: perform additional pointer sanity checks

### PR DESCRIPTION
**Pull request by slawblauciak**
Issues detected by code scans.

Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

**Comment by slawblauciak** [comment]
No, they don't.

File: src/audio/kpb.c [hunk 1]
```diff
@@ -641,13 +643,15 @@ static int kpb_copy(struct comp_dev *dev)
...
-		buffer_lock(sink, &flags);
+		if (sink)
+			buffer_lock(sink, &flags);
 
 		/* Validate sink */
 		if (!sink || !sink->stream.w_ptr) {
 			comp_err(dev, "kpb_copy(): invalid selector sink pointers.");
 			ret = -EINVAL;
-			buffer_unlock(sink, flags);
+			if (sink)
```
Comment (lgirdwood) [inline src/audio/kpb.c:653]: Isn't this checked above ? We should try and minimise this and fail if it's NULL (and we expect non NULL)

File: src/audio/kpb.c [hunk 2]
```diff
@@ -641,10 +647,16 @@ static int kpb_copy(struct comp_dev *dev)
 		/* In normal RUN state we simply copy to our sink. */
 		sink = kpb->sel_sink;
 
+		if (!sink) {
+			comp_err(dev, "kpb_copy(): no sink.");
+			ret = -EINVAL;
+			goto out;
```
Comment (paulstelian97) [inline src/audio/kpb.c:653]: Missing buffer_unlock(source, flags)?

Comment (slawblauciak) [reply src/audio/kpb.c:653]: It's unlocked before the switch.

**Comment by jajanusz** [comment]
@slawblauciak which code scan did you use? Clang analyzer? I wanted to start fixing this stuff once it will be plugged into GH, but if you are doing it already, then even better ;p

**Pull request merged**